overlord/hookstate/hookmgr.go

```diff
@@ -89,9 +90,14 @@ func Manager(s *state.State) (*HookManager, error) {
 }
 
 // HookTask returns a task that will run the specified hook.
-func HookTask(s *state.State, taskSummary, snapName string, revision snap.Revision, hookName string) *state.Task {
+func HookTask(s *state.State, taskSummary, snapName string, revision snap.Revision, hookName string, payload interface{}) *state.Task {
```

Comment:
@kyrofa I would find a way to express preseeding hook-context in the task, then Context.Get and Set can be used to access that, otherwise indeed we need yet another unflexible concept like payload, but maybe I'm missing the intention of the hook-context/Context.Get/Set stuff, afaiu it still not used atm